### ISSUE TITLE
[data_EDA]전체데이터 탐색 및 모델링 적용 

### ISSUE THREAD

**Issue by parkseonga** [DSC-SCH/livestock-solution-project#7]
전체 데이터 탐색은 상관분석만 진행하고 모델링을 적용해보았습니다. RF와 Xgboost를 적용하였는데 꽤 괜찮은 성능이 나왔습니다. 확인하시고 피드백 부탁드려요! 
[결과](https://github.com/DSC-SCH/livestock-solution-project/blob/data_EDA/eda/all_merged_data.ipynb)

전에 말씀드린 돼지탕박경매금액_주간평균 전처리에 관한 내용은 교수님의 자문을 구하고자합니다. 구체적인 내용은 회의 때 말씀드리겠습니다.

**Comment by NDjust** [comment]
@parkseonga 사진 깨졌어요!

**Comment by iamcodingcat** [comment]
수고하셨습니다! :) 모델링에 대한 피드백과 추가적인 안건은 회의 때 공유하겠습니다!

**Comment by NDjust** [comment]
@parkseonga @young-hun-jo 
우선 너무 고생하셨습니다.

내용을 봤는데, 한가지 의문이 들어 의견 남깁니다. 

- 우선 해당 데이터가 시계열 데이터인데, 왜 kfold로 평가를 진행 했는지
   - 시계열은 연속된 데이터라 random으로 train, test 을 나누면 잘못된 평가가 될 수 있다고 알고 있습니다.

- train, test로 평가 진행시 시간에 따라 나눠서 하는게 맞지 않을까요?


- 혹시 제가 틀린 부분이나 잘못 이해하는 부분이 있으면 말씀해주세요!

**Comment by iamcodingcat** [comment]
나단님께서 지적해주신 부분이 맞는 부분인 거 같습니다. 해당사항에 대해 구글링 검색해본 결과
시계열 데이터에 대한 Cross validation 방법이 따로 존재했고 확실히 저희가 기존에 사용했던 Kfold방법과는 다른 방법의 시계열 전용 Cross validation 방법이 있는 듯 합니다. 다행히 sklearn 패키지에 timeseries_split 이라는 매서드가 존재하는 것 같고 시계열 데이터에 맞는 방법으로 다시 해보겠습니다! 해당 메서드에 대한 링크 하단에 있습니다!
https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.TimeSeriesSplit.html

**Comment by NDjust** [comment]
@parkseonga @young-hun-jo  
그리고 추가로 Metric 기준을 어떤 걸로 할지 딱 정하고 하면 좋을 거 같아요.

RMSE 기준으로 볼 건지 아니면 해당 모형의 Accuracy라면 모형의 Accuracy가 나온 결과 값이 어떻게 구해졌는지 확인하고 선정해야할 것 같습니다.

**Comment by iamcodingcat** [comment]
혹시 Metric이 RMSE나 MAE와 같은 error관련된 수치는 바람직한 성능을 보이는데 Accuracy(regression이니 R2_score)값은 좋지 않게 나온다면 어떤 Metric을 선택할지 결정을 해야 할까요? 아니면 모든 Metric을 다 사용했을 때 모두 바람직한 결과가 나와야만 할까요? 의견 부탁드립니다!

**Comment by NDjust** [comment]
- https://towardsdatascience.com/regression-an-explanation-of-regression-metrics-and-what-can-go-wrong-a39a9793d914
- https://jihongl.github.io/2017/09/16/Rsquared/#:~:text=R%2Dsquared%EB%8A%94%20%EC%9A%B0%EB%A6%AC%EB%A7%90%EB%A1%9C%EB%8A%94,2%20%EC%9C%BC%EB%A1%9C%EB%8F%84%20%ED%91%9C%ED%98%84%EB%90%9C%EB%8B%A4.&text=%ED%9A%8C%EA%B7%80%EB%B6%84%EC%84%9D%EC%9D%80%20%EC%9C%84%EC%B2%98%EB%9F%BC,%EA%B0%80%20R%2Dsquared%EC%9D%B8%20%EA%B2%83%EC%9D%B4%EB%8B%A4.

R2 Score는 해당 변수들이 실제 값들을 얼마나 잘 설명하느냐를 보여주는 지표 아닌가요?? 

~~~
R-squared는 연구에 사용된 독립변수가 얼마나 적합한지 판단할 때 사용하여야 하고, 학습된 모델의 적합성을 F검정을 비롯한 다른 방법을 통해 얻어야 한다.
~~~

라는 내용이 있네요. 저는 정확히는 모르겠지만 사용 모델이 따라도 좀 다를 것 같고, MAE에서는 성능이 엄청 좋았는데, R-2에서는 성능이 너무 별로다라면 문제가 되겠는데 그런 게 아니라면 괜찮을 거 같아요!


+ 교수님께 자문을 구해보는게 어떨까요??
